### ISSUE TITLE
ref error when using "as" prop in <Visibility /> component

### ISSUE THREAD

**Issue by Nickersoft** [Semantic-Org/Semantic-UI-React#2412]
### Steps
I'm attempting to combine the <Visibility /> element with a <Grid.Row /> using the following:

```
    state = {
        visible: false
    };

    onUpdate(e, { calculations }) {
        if (this.state.visible === false) {
            this.setState({
                visible: calculations.topVisible
            });
        }
    }

    render() {
        const { header, description, size, image } = this.props;
        const { visible } = this.state;
        return (
            <Visibility as={Grid.Row} verticalAlign="middle" className={Styles.row} onUpdate={::this.onUpdate}>
                <Grid.Column>
                    <div
                        style={!visible ? { opacity: 0 } : null}
                        className={classnames(Styles.textWrapper, { [`${Animate.fadeInUp}`]: visible })}>
                        <Header as="h2" size="huge" className={Styles.listHeader}>
                            {header}
                        </Header>
                        <p className={Styles.listDescription}>{description}</p>
                    </div>
                </Grid.Column>
                <Grid.Column>
                    <div
                        style={!visible ? { opacity: 0 } : null}
                        className={classnames(Styles.imageWrapper, { [`${Animate.fadeInUp}`]: visible })}>
                        <Image centered size={size} src={image} />
                    </div>
                </Grid.Column>
            </Visibility>
        );
    }
```

### Expected Result
Should run as expected, fading in each row as it scrolls into view

### Actual Result
I'm receiving an error where it doesn't seem the "ref" is being set correctly:

```
Uncaught TypeError: Cannot read property 'getBoundingClientRect' of undefined
```

which seems to be thrown [here](https://github.com/Semantic-Org/Semantic-UI-React/blob/master/src/behaviors/Visibility/Visibility.js#L326).

This doesn't occur when the <Visibility /> is just the default `div`, however, in this case, it needs to be a row in order to format correctly. 

Can anyone offer any insight?

### Version
0.77.1

### Testcase
Seeing it's only one grid row, you'll have to resize the browser window to trigger the <Visibility /> component. The error should appear, though:

https://codesandbox.io/s/0m70jx1pnn

**Comment by layershifter** [comment]
Duplicate of #1321. `Visibility` tries to call `getBoundingClientRect()` on it's ref, while when `as` will receive a component its ref will be an object. Will be solved with #2306 and its successors.

I've made an example with workaround, [codesandbox](https://codesandbox.io/s/o7w8pln2p9).

**Comment by Nickersoft** [comment]
Hey @layershifter, thanks :) I had a feeling this issue was too commonplace to not be a duplicate, but searching beforehand I couldn't find anything. Will track the related issues.
